grype/presenter/models/source.go

```diff
@@ -14,6 +15,21 @@ type source struct {
 // newSource creates a new source object to be represented into JSON.
 func newSource(src syftSource.Description) (source, error) {
 	switch m := src.Metadata.(type) {
+	case pkg.PURLFileMetadata:
+		return source{
+			Type:   "purl-file",
+			Target: m.Path,
+		}, nil
+	case pkg.PURLLiteralMetadata:
+		return source{
+			Type:   "purl",
+			Target: m.PURL,
+		}, nil
+	case pkg.CPELiteralMetadata:
```

Comment:
Is this missing `CPEFileMetadata`?

Reply:
What about https://github.com/anchore/grype/pull/2463/files#diff-dabafbd923424ab88771650a65b50e02901d044e1466539f0e9d5fd61ed96dcbR42 `decodeCPEFile`?

Reply:
See the purl processing for this -- it treats cli input and file input the same way (so there is common parsing logic shared). I started with the same approach here, intending to add file support, but at the last second decided against it for simplicity. We can always add file support later, but when we do the shared logic is already in place (this function).

Point taken though on the name being confusing. I'll rename this function and the purl function to just hint they are parsing entries from a reader instead of a file.